xwiki-platform-core/xwiki-platform-oldcore/src/main/java/org/xwiki/internal/migration/AbstractDocumentsMigration.java

```diff
@@ -98,25 +104,37 @@ private Optional<XWikiDocument> convert(String documentReference)
     /**
      * @return the list of document ids to migrate
      */
-    protected abstract List<String> selectDocuments() throws DataMigrationException;
+    protected abstract List<ReferenceWithLocale> selectDocuments() throws DataMigrationException;
```

Comment:
A new @since could be interesting since, while the signature did not technically change, what is expected the List to contain did.